### PR TITLE
Add support for timemachine server

### PR DESCRIPTION
**Pull request by jgoerz**
This is very basic.  It only includes support for one user/share/password.

### CODE REVIEW

File: docs/applications/timemachine.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# Time Machine
+
+Apple docs: [https://support.apple.com/en-us/HT201250](https://support.apple.com/en-us/HT201250)
+Docker image: [https://github.com/mbentley/docker-timemachine](https://github.com/mbentley/docker-timemachine)
+
+Time Machine is an application that allows you to backup files from you Mac.
```
Comment (davestephens) [inline docs/applications/timemachine.md]: Please can you fix the typo in this line

File: docs/applications/timemachine.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+## Usage
+
+Set `timemachine_enabled: true` in your `group_vars/all.yml` file.
+
+Set the Time Machine specific variables:
```
Comment (davestephens) [inline docs/applications/timemachine.md]: All of this stuff should be under a "specific configuration" header - see https://github.com/davestephens/ansible-nas/blob/master/docs/applications/miniflux.md as an example.

File: group_vars/all.yml.dist [hunk 3]
```diff
@@ -181,6 +182,15 @@ samba_shares:
     browsable: yes
     path: "{{ samba_shares_root }}/photos"
 
+###
```
Comment (davestephens) [inline group_vars/all.yml.dist]: Please can you move this whole section down to the "you shouldn't need to edit..." section, then document as per Miniflux

File: group_vars/all.yml.dist [hunk 4]
```diff
@@ -181,6 +182,15 @@ samba_shares:
...
+### Time Machine
+###
+timemachine_data_directory: /mnt/Volume3/timemachine # needs to be an absolute path
```
Comment (davestephens) [inline group_vars/all.yml.dist]: For consistency, please change `timemachine_data_directory` to  `{{ docker_home }}/timemachine` by default.

File: tasks/timemachine.yml [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+    state: directory
+
+- name: Time Machine Docker Container
+  docker_container:
+    name: timemachine
+    image: mbentley/timemachine
+    pull: true
+    ports:
+      - "548:548"
+      - "5353:5353"
```
Comment (davestephens) [inline tasks/timemachine.yml]: What is this port for?

File: tests/test.yml [hunk 6]
```diff
@@ -182,6 +182,16 @@ samba_shares:
     browsable: yes
     path: "{{ samba_shares_root }}/photos"
 
+###
```
Comment (davestephens) [inline tests/test.yml]: Please move and update this section as per `all.yml.dist`

File: tasks/timemachine.yml [hunk 7]
```diff
@@ -0,0 +1,26 @@
+---
+- name: Time Machine Directories
+  file:
+    path: "{{ timemachine_data_directory }}"
+    state: directory
+
+- name: Time Machine Docker Container
+  docker_container:
+    name: timemachine
+    image: mbentley/timemachine
```
Comment (davestephens) [inline tasks/timemachine.yml]: Please can you specify `:latest` for the Docker image

File: tasks/timemachine.yml (hunk 5 above)
Comment (jgoerz) [reply tasks/timemachine.yml]: 5353 is for mDNS.  I retested and it doesn't seem to be required.  Removed.

File: docs/applications/timemachine.md (hunk 1 above)
Comment (jgoerz) [reply docs/applications/timemachine.md]: Done.

File: docs/applications/timemachine.md (hunk 2 above)
Comment (jgoerz) [reply docs/applications/timemachine.md]: Done.

File: group_vars/all.yml.dist (hunk 3 above)
Comment (jgoerz) [reply group_vars/all.yml.dist]: Done.

File: group_vars/all.yml.dist (hunk 4 above)
Comment (jgoerz) [reply group_vars/all.yml.dist]: Done.

File: tests/test.yml (hunk 6 above)
Comment (jgoerz) [reply tests/test.yml]: Done.

File: tasks/timemachine.yml (hunk 7 above)
Comment (jgoerz) [reply tasks/timemachine.yml]: Done.

**Comment by jgoerz** [comment]
Let me know if you want any other tweaks.  Ready for a review.

**Comment by davestephens** [comment]
Excellent, thanks!

**Pull request merged**